File: hw/pastor/u3/hw28PrinciplesAplication/JAVASCRIPT/model/genericEntity.js
class GenericEntity {
    constructor(type) {
        this._type = type;
        this._id = null;
        this._data = {}; 
    }

    setData(key, value) {
        this._data[key] = value;
    }

    getData(key) {
        return this._data[key];
    }

    static fromDocument(doc, type) {
        const entity = new GenericEntity(type);
        entity._id = doc._id;
        for (let key in doc) {
            if (key !== '_id') {
                entity.setData(key, doc[key]);
            }
        }
        return entity;
    }

    toDocument() {
        return { ...this._data };
    }
}

module.exports = GenericEntity;

File: hw/pastor/u3/hw28PrinciplesAplication/PYTHON/model/generic_entity.py
class GenericEntity:
    def __init__(self, type_name):
        self._id = None
        self._type = type_name
        self._data = {} 

    def set_data(self, key, value):
        self._data[key] = value

    def get_data(self, key):
        return self._data.get(key)
    
    def get_all_data(self):
        return self._data
    
    def set_id(self, id_val):
        self._id = id_val
        
    def get_id(self):
        return self._id

    def to_dict(self):
        doc = self._data.copy()
        if self._id:
            doc["_id"] = self._id
        return doc